src/com/walmartlabs/lacinia/pedestal/internal.clj

```diff
@@ -161,24 +186,25 @@
                               constants/parsed-query-key q))))
 
 (defn on-enter-query-excecutor
-  [context]
-  (let [resolver-result (execute-query context)
-        *result (promise)]
-    (resolve/on-deliver! resolver-result
-                         (fn [result]
-                           (deliver *result result)))
-    (apply-result-to-context context @*result)))
+  [interceptor-name stage]
+  (fn [context]
+    (let [resolver-result (execute-query context)
+          *result (promise)]
+      (resolve/on-deliver! resolver-result
+                           (fn [result]
+                             (deliver *result result)))
+      (apply-result-to-context context @*result interceptor-name stage))))
 
 (defn on-enter-async-query-executor
-  [context]
-  (let [ch (chan 1)
-        resolver-result (execute-query context)]
-    (resolve/on-deliver! resolver-result
-                         (fn [result]
-                           (->> result
-                                (apply-result-to-context context)
-                                (put! ch))))
-    ch))
+  [interceptor-name stage]
+  (fn [context]
+    (let [ch (chan 1)
+          resolver-result (execute-query context)]
+      (resolve/on-deliver! resolver-result
+                           (fn [result]
+                             (let [context (apply-result-to-context context result interceptor-name stage)]
```

Comment:
I don't quiet see the use of the `let` here, it could just be inline. That would also avoid re-defining `context`, which is something I tend to avoid.

Reply:
Oh; didn't even see that I was shadowing something! Fixed!